posts/2021-10-01-ga-release-post.adoc

```diff
@@ -0,0 +1,170 @@
+* Highlights of new features in Java 17 are found in the link:https://openjdk.java.net/projects/jdk/17/[Java 17 features page]
+* More detailed release information can be seen in the link:https://jdk.java.net/17/release-notes[release notes for Java 17]
+* Peruse link:https://docs.oracle.com/en/java/javase/17/docs/api/index.html[Javadocs for Java 17]
+
+For migration to Java 17, please check out link:https://blogs.oracle.com/javamagazine/migrate-to-java-17[Oracle Java 17 migration blog] and link:https://docs.oracle.com/en/java/javase/17/migrate/getting-started.html[Getting started with JDK 17].
+
+For Java 17 release roadmaps and availability refer to the link:https://adoptium.net/support.html?variant=openjdk17&jvmVariant=hotspot[Adoptium release roadmap] and link:https://www.oracle.com/java/technologies/java-se-support-roadmap.html[Oracle commercial JDK roadmap].
+
+And finally, for a comprehensive discussion on Java modules with a little hands on practice, read link:https://www.baeldung.com/java-9-modularity[this article from Baeldung].
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A210010+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.10].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18393[SPNEGO does not work with Java 11 or higher]
++
+In the release of Open Liberty 21.0.0.9, a regression was introduced which prevented the `Krb5LoginModule` class from being found when authenticating using Kerberos on Java 11 or higher. This issue caused some users to be presented with an `Error 403` or `Authentication Failed` when authenticating using the `spnego-1.0` feature. This issue has been fixed by reverting the underlying code back to the way it was prior to it being modified. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18559[Liberty is vulnerable to Information Disclosure (CVE-2021-29842 CVSS 3.7)]
++
+In the 21.0.0.9 release, an Information Disclosure vulnerability was discovered which could cause sensitive information to unintentionally be revealed. This issue has now been fixed in Open Liberty 21.0.0.10.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18475[Servlet ReadListener does not receive all HTTP request data]
++
+In 21.0.0.9 an issue was discovered where in rare cases, an app's `javax.servlet.ReadListener` could be invoked with `onAllDataRead()` prematurely, before all of the bytes sent along with the request have been made available to the app. In this scenario, `onDataAvailable()` would have been incorrectly invoked after the premature `onAllDataRead()` to provide notification for any remaining request data. This issue has been fixed by setting the `HttpServiceContextImpl` body to complete later on in the code.
```

Comment:
I don't think we should start this with `In 21.0.0.9`, that implies to me that this wasn't an issue in earlier releases. Suggested edit:

An issue was discovered where in rare cases an app's `javax.servlet.ReadListener` could be invoked with `onAllDataRead()` prematurely, before all of the bytes sent along with the request have been made available to the `ReadListener`. In an affected scenario, `onDataAvailable()` would be incorrectly invoked after the premature `onAllDataRead()` to provide notification for any remaining request data. This issue has been fixed with an update to the HTTP implementation which prevents HTTP request bodies from being prematurely marked as complete.